lib/Basic/PlatformUtility.cpp

```diff
@@ -348,7 +348,11 @@ std::string sys::makeTmpDir() {
   CreateDirectoryW((LPCWSTR)wPath.data(), NULL);
   return std::string(path);
 #else
-  char tmpDirPathBuf[] = "/tmp/fileXXXXXX";
+  char *tmpDirPathBuf = nullptr;
+  if (const char *tmpDir = std::getenv("TMPDIR"))
+    asprintf(&tmpDirPathBuf, "%s/fileXXXXXX", tmpDir);
```

Comment:
The buffer allocated here needs to be `free()`d before we return.

Reply:
Done.